app/models/workarea/catalog/product.decorator

```diff
@@ -1,7 +1,33 @@
 module Workarea
   decorate Catalog::Product, with: 'affirm' do
     decorated do
-      field :affirm_available, type: Boolean, default: true
+      field :affirm_available, type: Boolean
+      validate :gift_card_cannot_be_purchased_using_affirm
+    end
+
+    # The default value for `:affirm_available` depends on whether the
+    # Gift Cards plugin is installed, and `:gift_card` is set to true.
+    #
+    # @return [Boolean] Default value for the `:affirm_available` field.
+    def __affirm_available_default__
```

Comment:
This assumes this underscored version won't change in Mongoid - any reason not to use the default lambda which isn't fragile to Mongoid deciding they want to change this?